### PR TITLE
New pr for scan slice.

### PR DESCRIPTION
**Pull request by szyhf**
Update implement for suggestion in #473

### CODE REVIEW

**Comment by szyhf** [comment]
I don't know what's wrong with travis-ci's log:

```
../deps/jemalloc/lib/libjemalloc.a(nstime.o): In function `nstime_get':
/home/travis/gopath/src/gopkg.in/redis.v5/testdata/redis/deps/jemalloc/src/nstime.c:120: undefined reference to `clock_gettime'
collect2: ld returned 1 exit status
make[2]: *** [redis-server] Error 1
make[2]: Leaving directory `/home/travis/gopath/src/gopkg.in/redis.v5/testdata/redis/src'
make[1]: *** [all] Error 2
make[1]: Leaving directory `/home/travis/gopath/src/gopkg.in/redis.v5/testdata/redis'
make: *** [testdata/redis/src/redis-server] Error 
```

I didn't edit this file? @vmihailenco

File: internal/proto/scan.go [hunk 1]
```diff
@@ -105,3 +106,51 @@ func Scan(b []byte, v interface{}) error {
...
+		}
+	} else {
+		// is container of object
+		for index, s := range sSlice {
+			elem := internal.SliceNextElem(val)
+			if err := Scan([]byte(s), elem.Addr().Interface()); err != nil {
+				return fmt.Errorf("redis: ScanSlice failed at index of %d => %s, %s", index, s, err.Error())
+			}
+		}
+	}
```
Comment (vmihailenco) [inline internal/proto/scan.go]: Both branches (if/else) look the same... Can we remove one of them?

File: internal/proto/scan.go [hunk 2]
```diff
@@ -105,3 +106,51 @@ func Scan(b []byte, v interface{}) error {
...
+	// Check the if the container is pointer
+	if val.Kind() != reflect.Ptr {
+		return fmt.Errorf("redis: ScanSlice(non-pointer %T)", container)
+	}
+
+	// Is a valid value
+	val = val.Elem()
+	if !val.IsValid() {
+		return fmt.Errorf("redis: ScanSlice(non-pointer %T)", container)
+	}
```
Comment (vmihailenco) [inline internal/proto/scan.go]: What do you check here? I am not sure this is needed...

**Comment by vmihailenco** [comment]
Sorry for long delay (again). Please see my comments and rebase on master to fix the build.

>I didn't edit this file?

This is an issue in Redis build process which I fixed few days ago.

File: internal/proto/scan.go (hunk 1 above)
Comment (szyhf) [inline internal/proto/scan.go]: Sure ...... I don't know what did I do, lol

File: internal/proto/scan.go (hunk 2 above)
Comment (szyhf) [inline internal/proto/scan.go]: I don't know, too …… I felt like someone stolen my head ……

**Pull request closed without merging**